### ISSUE TITLE
Numput missing parameter

### ISSUE THREAD

**Issue by fznhq** [mmikeww/AHK-v2-script-converter#123]
input: 
`NumPut(1, NumPut(2, test, "int"), "int")`

output:
`NumPut("int", 1, test)`

expected output:
`NumPut("int", 2, "int", 1, test)`

**Comment by mmikeww** [comment]
In v2, NumPut is changed to allow writing multiple entries into memory at one time, as in your 'expected output' example. 

In v1, you can only write one item to memory at a time, but your input example tries to work around this by relying on the return value address and nesting two calls inside each other. I don't think the converter can handle this. We'd end up needing to write an entire expression parser to be able to check the return values of all functions.

**Comment by fznhq** [comment]
Thanks for reply,
yea i know it's _almost imposible_ to make parser from other function, but in this case only NumPut, not other functions, as the example I already gave, there is literally a missing value from the '**_main_**' NumPut

Other example:

input: 
`NumPut(1, NumPut(2,  NumPut(3,  NumPut(4, test, "int"), "int"), "int"), "int")`

converter output:
`NumPut("int", 3, "int", 2, "int", 1, test)`

see what i mean ?, the number 4 missing from output, it should output like this:
`NumPut("int", 4, "int", 3, "int", 2, "int", 1, test)`

**Comment by mmikeww** [comment]
I wasn't aware that NumPut already tried to handle this recursive case. It looks like @dmtr99 tried supporting it here in this commit https://github.com/mmikeww/AHK-v2-script-converter/commit/527b30932be7f2ff07a8fc935c137ea8c2f8a037 however unfortunately there were no tests added alongside. A day later it looks like he added this test (I would prefer that all tests get added in the same commit so its easier to find):

https://github.com/mmikeww/AHK-v2-script-converter/blob/master/tests/Test_Folder/External%20Libraries/NumputInNumput_WithContinuationSection.ah1

Since the converter already attempts to recursively unravel this, perhaps it is indeed fixable. I will take a look when I get some time, unless someone else does as well

**Comment by fznhq** [comment]
I think the problem occurs because of the way this code is placed

https://github.com/mmikeww/AHK-v2-script-converter/blob/5c05902e1f22cb6f933e55a18d9116369eef01e2/convert/2Functions.ahk#L264

that code placed after this checks whether there is another NumPut

https://github.com/mmikeww/AHK-v2-script-converter/blob/5c05902e1f22cb6f933e55a18d9116369eef01e2/convert/2Functions.ahk#L259-L262

then if there is no other NumPut, then the '**_main_**' value got ignored, because the loop already stopped

basically the solution is just move the ParBuffer assigment before the last check, i can help if you want

**Comment by mmikeww** [comment]
feel free to submit a pull request if you solve it. 

make sure you add a few test files alongside
